### PR TITLE
fix Response.statusText

### LINKED ISSUE #5753: Response.statusText is empty
```js
console.log(
    new Response("foo", {
        statusText: "ok",
        status: 200
    }).statusText
);
```

Appears empty, seems like a bug?

_Originally reported on Discord: [`Ask Inkeep`](https://discord.com/channels/876711213126520882/1153742219480662068/1153742220948680834)_

### PR DESCRIPTION
**Pull request by toshok**
### What does this PR do?

Restructures `Response`/`Body` a little, moving `Body.Init` under `Response`.  Also adds `statusText` handling to `Response.Init` (which fixes the linked bug).  This had a few other knock-on effects, like moving `@"200"`/`@"404"` out of `Body` (again to `Response`.)

fixes #5753

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

I wrote a some tests to handle various cases.

### CODE REVIEW

File: src/bun.js/bindings/bindings.cpp [hunk 1]
```diff
@@ -4114,6 +4115,9 @@ static JSC::Identifier builtinNameMap(JSC::JSGlobalObject* globalObject, unsigne
     case BuiltinNamesMap::headers: {
         return clientData->builtinNames().headersPublicName();
     }
+    case BuiltinNamesMap::statusText: {
+        return clientData->builtinNames().statusTextPublicName();
+    }
```
Comment (toshok) [inline src/bun.js/bindings/bindings.cpp:4128]: all of the bindings{.cpp/.zig} changes were very much dogscience.  are there tests or things I should worry about?

**Comment by toshok** [comment]
a lot of the code motion was due to me trying to make things clearer for myself while I fixed the bug.  I could open a different PR that only addresses the bug and save the code motion for after, if you'd prefer.  lemme know.

File: src/bun.js/webcore/response.zig [hunk 2]
```diff
@@ -172,7 +178,7 @@ pub const Response = struct {
     }
 
     pub fn isOK(this: *const Response) bool {
-        return this.body.init.status_code == 304 or (this.body.init.status_code >= 200 and this.body.init.status_code <= 299);
+        return this.init.status_code == 304 or (this.init.status_code >= 200 and this.init.status_code <= 299);
```
Comment (Electroid) [inline src/bun.js/webcore/response.zig]: Unrelated to your changes, but might as well fix it. `response.ok` is only suppose to be true for `[200,300)`. This was a mistake when implemented initially.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ Today (February 2023), Bun's codebase has five distinct parts:
 
 - JavaScript, JSX, & TypeScript transpiler, module resolver, and related code
 - JavaScript runtime ([`src/bun.js/`](src/bun.js/))
-- JavaScript runtime bindings ([`src/bun.zig/bindings/**/*.cpp`](src/bun.zig/bindings/))
```
Comment (Jarred-Sumner) [inline CONTRIBUTING.md]: nice

File: src/bun.js/bindings/bindings.cpp (hunk 1 above)
Comment (Jarred-Sumner) [reply src/bun.js/bindings/bindings.cpp:4128]: this is right, thanks

context here: `Identifier::fromString` is super slow. It frequently shows up in profiles. It has a lock, it clones a string and hashes it each call

**Review by Jarred-Sumner — APPROVED** [review]
Looks good. I'm pretty confident we can merge it, but lets let the tests run. There will be a handful of unrelated test failures, mostly due to permissions but also because CI isn't currently green (especially on macOS)

**Comment by Jarred-Sumner** [comment]
Looks like there are some new test failures

**Comment by toshok** [comment]
☝️ rebased off main to clear up the conflict.  incoming two commits fix the test failures (`js/bun/http` passes now) and addresses the suggested change above.

File: src/bun.js/webcore/response.zig (hunk 2 above)
Comment (toshok) [reply src/bun.js/webcore/response.zig]: done

**Comment by Jarred-Sumner** [comment]
Don't worry about the `cp.test` failure, fixed in 54d25b8f6b0f30b07146162a15d5f500fd45b2ee

**Comment by Jarred-Sumner** [comment]
`inspect.test` needs to be updated

**Comment by paperclover** [comment]
image

Ordering of these fields changed (inspect.test.js). I think it's okay if these tests are updated.

**Comment by Electroid** [comment]
Thanks for working on this!

**Pull request merged**